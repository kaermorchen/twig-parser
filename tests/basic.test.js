import { test, expect } from 'vitest';
import Parser from '../out/parser.js';

test('Numbers', () => {
  expect(Parser.parse(`{{ 42 }}`)).toMatchSnapshot();
});

test('Strings', () => {
  expect(Parser.parse(`{{ "Hello" }}`)).toMatchSnapshot();
});

test('Identifier', () => {
  expect(Parser.parse(`{{ foo }}`)).toMatchSnapshot();
});
