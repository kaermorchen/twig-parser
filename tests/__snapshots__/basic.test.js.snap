// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Identifier 1`] = `
{
  "body": [
    {
      "type": "Identifier",
      "value": "foo",
    },
  ],
  "type": "Block",
}
`;

exports[`Numbers 1`] = `
{
  "body": [
    {
      "type": "NumericLiteral",
      "value": 42,
    },
  ],
  "type": "Block",
}
`;

exports[`Strings 1`] = `
{
  "body": [
    {
      "type": "StringLiteral",
      "value": "Hello",
    },
  ],
  "type": "Block",
}
`;
